Doc/mdbook (#187)
Wenzel/libmicrovmi: doc/src/tutorial/c.md

# Memory-dump example in C

In this tutorial chapter, we will run the `mem-dump.c` C example code

> to dump a domain physical memory.

Reviewer: same here

First, we will compile libmicrovmi with the Xen driver enabled:

~~~
$ cd libmicrovmi
$ cargo build --features xen
~~~

## Install cbindgen

> [`cbindgen`](https://github.com/eqrion/cbindgen) is an automated C bindings generation tool from Rust.

Reviewer: ```suggestion
[`cbindgen`](https://github.com/eqrion/cbindgen) is a tool for automatically generating C header files that expose a library's [FFI](https://doc.rust-lang.org/nomicon/ffi.html).
```

To install it:
~~~
$ cargo install --force cbindgen
~~~

## Building mem-dump C example

~~~
$ cd c_examples/
$ make mem-dump
~~~

## Running the example

Next, let's run the example on your domain `<vm_name>`:

~~~
$ sudo LD_LIBRARY_PATH="$LD_LIBRARY_PATH:../target/debug" ./mem-dump <vm_name>
~~~

⚠️ Note: `libmicrovmi.so` has been generated by cargo into `target/debug`.

> We have to update `LD_LIBRARY_PATH` so that `mem-dump` can link with the library.

Reviewer: ```suggestion
We have to set `LD_LIBRARY_PATH` so that `mem-dump` is able to find the library.
```
Linking is done at compile time ;)
